### ISSUE TITLE
Keyboard navigation glitch when using mouse to click outside injection div during field edit

### ISSUE THREAD

**Issue by microbit-matt-hillsdon** [microsoft/pxt-microbit#6461]
**Describe the bug**

There's a keyboard navigation glitch when using mouse to click outside injection div during a field edit (i.e. with widget or dropdown focused) and then tabbing back to the toolbox or workspace.

In this scenario keyboard navigation doesn't work properly.

If you click in the workspace, use the area menu or close the toolbox with escape after tabbing to it everything works again.

We tried to fix this for the 12.2.0 Blockly release but it introduced more serious issues for keyboard only users so we've had to back away from it for now.

**To Reproduce**
1. Edit a field but do not confirm/cancel (e.g. leave dropdown open or input focused)
2.  Click in whitespace below the simulator
3. Tab back to the toolbox
4. Try to access the flyout via keyboard, note it doesn't work
5. Click in the workspace or Escape to close toolbox, everything works again.

**Expected behavior**
We think the dropdown and widget should close when clicking outside the injection div, just like they do when clicking on empty space inside the workspace.

Blockly agree and tried to make a change for 12.2.0. Unfortunately this broke scenarios in MakeCode that use both the widget and dropdown div. See https://github.com/google/blockly/issues/9221 so the released 12.2.0 does not include this change.

There are potential workarounds in MakeCode, see the hideChaff suggestion under "Potential solutions". I don't think we should attempt them for Friday but potentially as a subsequent fix.
